ndk-glue/src/lib.rs

```diff
@@ -334,6 +338,9 @@ unsafe extern "C" fn on_stop(activity: *mut ANativeActivity) {
 unsafe extern "C" fn on_destroy(activity: *mut ANativeActivity) {
     wake(activity, Event::Destroy);
     ndk_context::release_android_context();
+    let mut native_activity_guard = NATIVE_ACTIVITY.write();
+    let native_activity = native_activity_guard.take().unwrap();
+    assert_eq!(native_activity.ptr().as_ptr(), activity);
```

Comment:
`NativeActivity` doesn't have a drop handler (that would previously call `.finish()` in the original revision of this PR) so nothing funky is going to happen, just that users will now get `None` out of `fn native_activity()`.